src/components/molecules/VenuePartygoers/VenuePartygoers.tsx

```diff
@@ -1,31 +1,39 @@
 import React, { useMemo } from "react";
 
-import { useSelector } from "hooks/useSelector";
-import { usePartygoers } from "hooks/users";
-
-import { currentVenueSelector, parentVenueSelector } from "utils/selectors";
+import { useRecentWorldsUsers } from "hooks/users";
+import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
+import { useVenueId } from "hooks/useVenueId";
 
 import "./VenuePartygoers.scss";
 
 export const VenuePartygoers = () => {
-  const venue = useSelector(currentVenueSelector);
-  const parentVenue = useSelector(parentVenueSelector);
-  const partygoers = usePartygoers();
-
-  const currentVenueTitle = venue?.attendeesTitle ?? "partygoers";
-  const attendeesTitle = parentVenue?.attendeesTitle ?? currentVenueTitle;
-  const venueName = venue?.name;
-
-  const currentVenuePartygoers = useMemo(() => {
-    if (!venueName) return [];
-
-    return partygoers.filter((partygoer) => partygoer.lastSeenIn?.[venueName]);
-  }, [partygoers, venueName]);
-  const numberOfPartygoers = currentVenuePartygoers.length;
-
-  return (
-    <div className="venue-partygoers-container">
-      {numberOfPartygoers} {attendeesTitle} online
-    </div>
-  );
+  const venueId = useVenueId();
+  const {
+    parentVenue,
+    currentVenue,
+    isCurrentVenueLoaded,
+  } = useConnectRelatedVenues({ venueId });
+
+  const { recentWorldUsers, isRecentWorldUsersLoaded } = useRecentWorldsUsers();
+
+  const numberOfRecentWorldUsers = recentWorldUsers.length;
+
+  const title = useMemo<string>(() => {
+    if (!isCurrentVenueLoaded || !isRecentWorldUsersLoaded) return "";
+
+    const attendeesTitle =
+      parentVenue?.attendeesTitle ??
+      currentVenue?.attendeesTitle ??
+      "partygoers";
```

Comment:
I'm not 100% here, but I think we may want a less 'party' feeling word here as our fall back default.. maybe just `attendees`? If there are examples elsewhere in the codebase, aligning to what they use would be good.